xayagame/Makefile.am

```diff
@@ -3,13 +3,27 @@ xayagamedir = $(includedir)/xayagame
 
 EXTRA_DIST = rpc-stubs/game.json rpc-stubs/xaya.json
 
+
 libxayagame_la_CXXFLAGS = \
   $(JSONCPP_CFLAGS) $(JSONRPCCLIENT_CFLAGS) $(JSONRPCSERVER_CFLAGS) \
   $(GLOG_CFLAGS) $(SQLITE3_CFLAGS) $(ZMQ_CFLAGS)
 libxayagame_la_LIBADD = \
   $(JSONCPP_LIBS) $(JSONRPCCLIENT_LIBS) $(JSONRPCSERVER_LIBS) \
   $(GLOG_LIBS) $(SQLITE3_LIBS) $(ZMQ_LIBS) \
   -lstdc++fs
+  
+
+# When building for windows, jsonrpccpp library can only be build without hredis;
+# So we need to make sure, the -lhredis flag is not passed
+# Also, lets pass -no-undefined to make sure we have zero undefined symbols
+# Else mingw will not be able to build shared libraries
+if WINDOWS
+  libxayagame_la_LDFLAGS = $(JSONCPP_LDFLAGS) $(JSONRPCCLIENT_LDFLAGS) $(JSONRPCSERVER_LDFLAGS) \
+  $(GLOG_LDFLAGS) $(SQLITE3_LDFLAGS) $(ZMQ_LDFLAGS) -no-undefined
```

Comment:
Is adding `-no-undefined` here related in any way to the `win32-dll` thing from above?  Do you need both?  Also there seems to be nothing related to `hredis` in the actual code, you just mention it in the comment.  Is this an oversight?

Reply:
I am not sure, I remember I only got correct results with combination of win32-dll being set there and -no-undefined later on. I will try removing those now and see, if that is really necessary. If so, so I will supply error log with explanations. As for the 'hredis', I resolved it by directly modifying SON-RPC-CPP source, and so its "resolved" outsided this project scope. They are going to merge my fix soon.

Reply:
Ok, thanks for fixing the hredis thing upstream!  Then just remove its mention from the comment, as it is misleading otherwise.

Reply:
"Then just remove its mention from the comment, as it is misleading otherwise." - good point, will do